controllers/issuer_controller_test.go

```diff
@@ -61,7 +62,13 @@ func TestSimpleIssuerReconcilerReconcile(t *testing.T) {
 		expectedEvents      []string
 	}
 
-	fakeTimeObj := metav1.NewTime(cmtime.FakeTime)
+	fakeTime1 := time.Now().Truncate(time.Second)
+	fakeTimeObj1 := metav1.NewTime(fakeTime1)
+	fakeClock1 := clocktesting.NewFakeClock(fakeTime1)
+
+	fakeTime2 := time.Now().Add(4 * time.Hour).Truncate(time.Second)
+	fakeTimeObj2 := metav1.NewTime(fakeTime2)
+	fakeClock2 := clocktesting.NewFakeClock(fakeTime2)
```

Comment:
Thanks, I think there definitely could be an issue like that in the future. However, I think that some tests might still succeed incorrectly when the date is set far in the past or far in the future. I think it might be useful to use a random date here, such a test will fail eventually, indicating that we aren't using the correct clock somewhere in code.

Reply:
see https://github.com/cert-manager/issuer-lib/pull/16/commits/642ff1721821861b708d77c41bb90a5ad5c8c623